pyfarm/models/jobqueue.py

```diff
@@ -197,7 +197,7 @@ def clear_assigned_counts(self):
         if self.parent:
             self.parent.clear_assigned_counts()
 
-    def get_job_for_agent(self, agent):
+    def get_job_for_agent(self, agent, unwanted_job_ids=[]):
```

Comment:
`unwanted_job_ids` should be None or some other value as a default.  The same list object will be reused each time `get_job_for_agent()` with this code.